statsbestcustomers.php

```diff
@@ -168,9 +168,10 @@ public function getData()
 				AND o.valid
 			), 0) as totalValidOrders
 		FROM `'._DB_PREFIX_.'customer` c
+        INNER JOIN `'._DB_PREFIX_.'orders` o ON o.id_customer = c.id_customer 
 		LEFT JOIN `'._DB_PREFIX_.'guest` g ON c.`id_customer` = g.`id_customer`
-		LEFT JOIN `'._DB_PREFIX_.'connections` co ON g.`id_guest` = co.`id_guest`
-		WHERE co.date_add BETWEEN '.$this->getDate()
+        LEFT JOIN `'._DB_PREFIX_.'connections` co ON g.`id_guest` = co.`id_guest` AND co.date_add BETWEEN '.$this->getDate().'
+		WHERE o.date_add BETWEEN '.$this->getDate()
```

Comment:
Not sure it's mandatory because the date_add could be not in the range of `getDate` contrary to the invoice date
:thinking: 
Maybe the INNER JOIN on orders is enough?

Reply:
I don't think so. The displayed stats are : visits/orders/ ... for a time range. Visits and orders are not related.

Reply:
Ping @PierreRambaud